types/mithril/index.d.ts

```diff
@@ -246,7 +246,7 @@ declare namespace Mithril {
 	 * Components are a mechanism to encapsulate parts of a view to make code easier to organize and/or reuse.
 	 * Any Javascript object that has a view method is a Mithril component. Components can be consumed via the m() utility.
 	 */
-	type Comp<Attrs, State extends Lifecycle<Attrs, State>> = Component<Attrs, State> & State;
```

Comment:
@isiahmeadows note that my example does not reassign the entire `state` object as discussed in https://github.com/MithrilJS/mithril.js/issues/1727; it only adds the `onremove` property to state.